src/Visitor/MutationsCollectorVisitor.php

```diff
@@ -66,6 +71,13 @@ public function leaveNode(Node $node)
                 }
             }
 
+            if ($mutator->getConfig()->isIgnored(
```

Comment:
1. Since `$mutator->getConfig()` must return an instance of `MutantConfig`, it's better to create Mutator with constructot injection: `new Mutator(new MutatorConfig())'`
Thus, it won't be possible to have invalid object.

2. Probably, we shouldn't expose `MutatorConfig` to client code. I mean, we can check whether Mutator is ignored or not like this

```php
class Mutator 
{
     public function __construct(MutatorConfig $mutatorConfig)
     {
          ...
     }

     public function isIgnored(): bool
     {
         return $this->config->isIgnored();
     }
}
```

so `$mutator->isIgnored()` instead of `$mutator->getConfig()->isIgnored()`